keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -1972,6 +2022,10 @@ exist at all must not be allowed to run. Instead, a suitable event must be
 emitted which explains the problem. Such a situation can occur as part of
 downgrade scenarios.
 
+In addition, kubelet informs about the claim status with a
+`ResourceClaimsReady` `PodConditionType`. It's `false` when NodePrepareResource
+still needs to be called. It's `true` when that has succeeded for all claims.
```

Comment:
Any thoughts on whether such a condition is useful? I'm undecided myself.

@klueska, @smarterclayton ?

Reply:
My thinking: it's all-or-nothing. So, if something is interested in a particular claim, it's not helpful. If you want the side effect of a metric changing to see how long it's taking Pods to get to this point, it might be.

Maybe we can provide another instrumentation approach though. For example, one or more of:
- increment a per-node counter when a Pod is accepted with pending claims, and increment a different counter when claims are resolved
- put the `.status` in the ResourceClaim instead, with an entry for each associated Pod;
  teach tooling to look there
- fire an Event [per claim, per Pod] to note that a resource claim has succeeded and is ready

Reply:
(thinking ahead)
Could we ever have a situation where a Pod that is already running might make a resource claim?
For example, a Pod that is using one cryptographic key identifies that it needs to make another (key rotation) and so requests a second one, by directly calling *create* for the ResourceClaim API.

In that case, we wouldn't expect to update `.status` for the Pod; we might not even have the associated Pod object identity available, depending on how it has authenticated.

Reply:
I removed the condition part. That needs more thought about use cases.

Reply:
One of the challenges on the debuggability today is the user / dev don't know why a pod being scheduled but not running. The scenario of an unallocated or unreserved claims makes the situation even more complicated. But I agreed this need more thoughts.